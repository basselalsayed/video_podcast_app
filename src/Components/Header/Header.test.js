import React from 'React';
import { Header } from './';

const rssHandler = jest.fn(() => {});

const defaultProps = {
  link: 'www.test.com',
  title: 'Test Title',
  rssHandler,
};

describe('<Header />', () => {
  const wrapper = shallow(<Header {...defaultProps} />);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has correct inner text and link', () => {
    expect(wrapper.text()).toEqual('Test TitleUpdate Feed');
    expect(wrapper.html()).toEqual(
      '<nav class="navbar navbar-expand navbar-light"><a id="brand-link" tabindex="0" href="www.test.com"><span class="navbar-brand">Test Title</span></a><button type="button" aria-label="Toggle navigation" class="navbar-toggler collapsed"><span class="navbar-toggler-icon"></span></button><div class="justify-content-end navbar-collapse collapse"><button id="refresh-rss" type="button" class="btn btn-primary">Update Feed</button></div></nav>',
    );
  });

  it('calls the reset spy onClick', () => {
    wrapper.find('#refresh-rss').simulate('click');
    expect(rssHandler.mock.calls.length).toEqual(1);
  });
});
