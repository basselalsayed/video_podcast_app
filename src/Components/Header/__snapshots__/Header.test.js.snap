// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<Fragment>
  <Navbar
    collapseOnSelect={false}
    expand={true}
    variant="light"
  >
    <NavbarBrand>
      <p>
        Test Title
      </p>
    </NavbarBrand>
    <NavbarToggle
      label="Toggle navigation"
    />
    <NavbarCollapse
      className="justify-content-end"
    >
      <Button
        active={false}
        disabled={false}
        id="refresh-rss"
        onClick={[MockFunction]}
        type="button"
        variant="primary"
      >
        Update Feed
      </Button>
    </NavbarCollapse>
  </Navbar>
</Fragment>
`;
